### PR TITLE
Added --default-image option to be used with --shell-executor-no-image=false

### PR DESCRIPTION
**Pull request by NGPetrov**
Currently if `--shell-executor-no-image=false` is used the runner defaults to `docker.io/ruby:3.1`.
In case isolation is required, there is no option to run the ci in a custom image.

Added --default-image option. 
If set when `--shell-executor-no-image=false`, the value will be used as a image source for the container.
If not set when `--shell-executor-no-image=false`, the default `docker.io/ruby:3.1` will be used.
If set when `--shell-executor-no-image=true` (or not set), warring message is displayed.
If set when `--shell-isolation`, warring message is displayed.

~Fixed typo (or old option name) in `arvg.ts:49`, from `--no-shell-executor-no-image` to  `--shell-executor-no-image`~

Included 3 tests to validate proper runner image/shell is used.

### CODE REVIEW

File: src/argv.ts [hunk 1]
```diff
@@ -46,8 +46,17 @@ export class Argv {
         argv.injectDotenv(`${argv.cwd}/.gitlab-ci-local-env`, args);
 
         if (!argv.shellExecutorNoImage && argv.shellIsolation) {
-            writeStreams?.stderr(chalk`{black.bgYellowBright  WARN } --shell-isolation does not work with --no-shell-executor-no-image\n`);
+            writeStreams?.stderr(chalk`{black.bgYellowBright  WARN } --shell-isolation does not work with --shell-executor-no-image\n`);
```
Comment (ANGkeith) [inline src/argv.ts:49]: not really a typo, 

https://yargs.js.org/docs/#api-reference-booleankey
`--no-shell-executor-no-image` is an alias for `--shell-executor-no-image=false`

File: src/argv.ts [hunk 2]
```diff
@@ -273,6 +282,10 @@ export class Argv {
         return this.map.get("shellExecutorNoImage") ?? true;
     }
 
+    get defaultImage (): string | null {
+        return this.map.get("defaultImage") ?? null;
```
Comment (ANGkeith) [inline src/argv.ts:286]: image
lets move them here instead

Comment (ANGkeith) [reply src/argv.ts:286]:
```
// https://docs.gitlab.com/ee/ci/runners/hosted_runners/linux.html#container-images
return this.map.get("defaultImage") ?? "docker.io/ruby:3.1";
```
yup, this is what i meant which is mostly how we've set the default args for all the arguments.

**Review by ANGkeith — APPROVED** [review]

**Review by firecow — APPROVED** [review]

File: package.json
Comment (firecow) [inline package.json:1]: What are these weird changes?

Comment (firecow) [reply package.json:1]: I'm talking about the specific noop changes to package.json and package-lock.json :smile:

File: package-lock.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
         "depcheck": "1.x.x",
         "esbuild": "^0.24.0",
         "eslint": "8.x.x",
-        "jest": "29.x.x",
+        "jest": "^29.7.0",
```
Comment (firecow) [inline package-lock.json:56]:
```suggestion
        "jest": "29.x.x",
```

**Pull request merged**